earth_enterprise/src/common/gedbroot/proto_dbroot_unittest.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include "proto_dbroot.h"
+#include "common/khSimpleException.h"
+
+// Test various error and non-error conditions when loading proto dbroots.
+// Since we're validating the logic for loading the dbroots, we don't worry too
+// much about the contents.
+
+TEST(ProtoDbroot, InvalidFile) {
```

Comment:
I started writing unit tests for the Eta dbroot as well, but it would take significant effort to find an Eta dbroot and it's not worth the effort for something that's never going to change.